client-react/src/pages/app/deployment-center/container/DeploymentCenterContainerAcrDataLoader.tsx

```diff
@@ -170,179 +166,170 @@ const DeploymentCenterContainerAcrDataLoader: React.FC<DeploymentCenterFieldProp
-            } else {
-              const errorMessage = getErrorMessage(credentialsResponse.metadata.error);
-              const statusMessage = errorMessage
-                ? t('deploymentCenterContainerAcrFailedToLoadCredentialsWithError').format(errorMessage)
-                : t('deploymentCenterContainerAcrFailedToLoadCredentials');
+    setLoadingImageOptions(true);
+    setAcrImageOptions([]);
+    setAcrTagOptions([]);
+    clearStatusBanner();
+    const serverUrl = loginServer?.toLocaleLowerCase() ?? '';
 
-              setAcrStatusMessage(statusMessage);
-              setAcrStatusMessageType(MessageBarType.error);
+    const selectedRegistryIdentifier = registryIdentifiers.current[serverUrl];
 
-              portalContext.log(
-                getTelemetryInfo('error', 'credentialsResponse', 'failed', {
-                  message: getErrorMessage(credentialsResponse.metadata.error),
-                  error: credentialsResponse.metadata.error,
-                })
-              );
-            }
-          }
+    if (selectedRegistryIdentifier) {
+      if (selectedRegistryIdentifier.adminUserEnabled) {
+        if (!selectedRegistryIdentifier.credential) {
+          portalContext.log(getTelemetryInfo('info', 'listAcrCredentials', 'submit'));
+          const credentialsResponse = await deploymentCenterData.listAcrCredentials(selectedRegistryIdentifier.resourceId);
 
-          const credentials = registryIdentifiers.current[serverUrl] ? registryIdentifiers.current[serverUrl].credential : undefined;
-
-          if (credentials) {
-            const username = credentials.username;
-            const password = credentials.passwords[0].value;
-            let failedNetworkCall = false;
-            let errorMessage = '';
-
-            portalContext.log(getTelemetryInfo('info', 'getAcrRepositories', 'submit'));
-
-            const repositoriesResponse = await deploymentCenterData.getAcrRepositories(
-              portalContext,
-              loginServer,
-              username,
-              password,
-              (page, response: IDataMessageResult<any>) => {
-                portalContext.log(
-                  // NOTE(michinoy): 2021-02-04, Generally a bad idea to log the entire response object. But I am unable to identify what error is being returned,
-                  // thus logging the entire response object.
-                  getTelemetryInfo('error', 'getAcrRepositoriesResponse', 'failed', {
-                    page: page,
-                    error: response.status,
-                  })
-                );
-
-                failedNetworkCall = !isPortalCommunicationStatusSuccess(response.status);
-                errorMessage = getErrorMessage(response.status);
-              }
+          if (credentialsResponse.metadata.success && credentialsResponse.data.passwords && credentialsResponse.data.passwords.length > 0) {
```

Comment:
`credentialsResponse.data.passwords` is declared as non-nullable.
```suggestion
          if (credentialsResponse.metadata.success && credentialsResponse.data.passwords.length > 0) {
```

Or update the `ACRCredential` type if it is nullable.

```ts
export interface ACRCredential {
  username: string;
  passwords?: ACRCredentialPassword[];
}
```